pkg/bundle/plain/v0/bundle.go

```diff
@@ -0,0 +1,382 @@
+		if ns == "" {
+			continue
+		}
+
+		role := *role
+		role.Namespace = ns
+		roles = append(roles, &role)
+	}
+
+	b.StoreObjects(role.Name+"_generated.yaml", role) // TODO(ryantking): Better name
+	return nil
+}
+
+func (b *Bundle) newRoleBindings(
+	name string,
+	permission operatorsv1alpha1.StrategyDeploymentPermissions,
+) error {
+	roleBindings := make([]*rbacv1.RoleBinding, 0, len(b.targetNamespaces))
+	obj, err := b.Scheme().New(rbacv1.SchemeGroupVersion.WithKind("RoleBinding"))
+	if err != nil {
+		return err
+	}
+
+	roleBinding := obj.(*rbacv1.RoleBinding)
+	roleBinding.Name = name
+	roleBinding.Subjects = []rbacv1.Subject{
+		{
+			Kind:      "ServiceAccount",
+			Name:      permission.ServiceAccountName,
+			Namespace: b.installNamespace,
+		},
+	}
+	roleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "Role",
+		Name:     name,
+	}
+
+	for _, ns := range b.targetNamespaces {
+		if ns == "" {
+			continue
+		}
+
+		roleBinding := *roleBinding
+		roleBinding.Namespace = ns
+		roleBindings = append(roleBindings, &roleBinding)
+	}
+
+	b.StoreObjects(roleBinding.Name+"_generated.yaml", roleBinding) // TODO(ryantking): Better name
+	return nil
+}
+
+func (b *Bundle) newClusterRoles(
+	name string,
+	permission operatorsv1alpha1.StrategyDeploymentPermissions,
+) error {
+	roles := make([]*rbacv1.ClusterRole, 0, len(b.targetNamespaces))
+	obj, err := b.Scheme().New(rbacv1.SchemeGroupVersion.WithKind("ClusterRole"))
+	if err != nil {
+		return err
+	}
+
+	role := obj.(*rbacv1.ClusterRole)
+	role.Name = name
+	role.Rules = permission.Rules
+	for _, ns := range b.targetNamespaces {
+		if ns == "" {
+			continue
+		}
+
+		role := *role
+		role.Namespace = ns
+		roles = append(roles, &role)
+	}
+
+	b.StoreObjects(role.Name+"_generated.yaml", role) // TODO(ryantking): Better name
+	return nil
+}
+
+func (b *Bundle) newClusterRoleBindings(
+	name string,
+	permission operatorsv1alpha1.StrategyDeploymentPermissions,
+) error {
+	roleBindings := make([]*rbacv1.ClusterRoleBinding, 0, len(b.targetNamespaces))
+	obj, err := b.Scheme().New(rbacv1.SchemeGroupVersion.WithKind("ClusterRoleBinding"))
+	if err != nil {
+		return err
+	}
+
+	roleBinding := obj.(*rbacv1.ClusterRoleBinding)
+	roleBinding.Name = name
+	roleBinding.Subjects = []rbacv1.Subject{
+		{
+			Kind:      "ServiceAccount",
+			Name:      permission.ServiceAccountName,
+			Namespace: b.installNamespace,
+		},
+	}
+	roleBinding.RoleRef = rbacv1.RoleRef{
+		APIGroup: rbacv1.GroupName,
+		Kind:     "ClusterRole",
+		Name:     name,
+	}
+
+	for _, ns := range b.targetNamespaces {
+		if ns == "" {
+			continue
+		}
+
+		roleBinding := *roleBinding
+		roleBinding.Namespace = ns
+		roleBindings = append(roleBindings, &roleBinding)
+	}
+
+	b.StoreObjects(roleBinding.Name+"_generated.yaml", roleBinding) // TODO(ryantking): Better name
+	return nil
+}
+
+func (b Bundle) generateName(base string, o interface{}) string {
+	const maxNameLength = 63
```

Comment:
nit: we enforce that the length of a rukpak Bundle type is maximum 40 characters long (see https://github.com/operator-framework/rukpak/blob/main/manifests/apis/crds/patches/bundle_validation.yaml#L1). Would it make sense to have a similar limit here?

Reply:
Pinging @joelanford and @timflannagan to weigh on this one.